BUILDING-cmake.md

```diff
@@ -263,7 +263,7 @@ documentation on how to set up the libprojectM build system for use in another p
 ### Importing libprojectM targets from an installed version
 
 This is the recommended and supported way of importing libprojectM in your project. This project installs a set of CMake
-files in `<PREFIX>/<LIBDIR>/cmake/libprojectM`, containing target definitions, version and dependency checks as well as
+files in `<PREFIX>/<LIBDIR>/cmake/projectM-4.0`, containing target definitions, version and dependency checks as well as
```

Comment:
When renaming the package this way, be aware that this will make `find_package` potentially unusable because it doesn't really like dots in the name. this would make CMake-based integration in other projects impossible.

I'd suggest the following, which other libs also do:
- Rename the current files/package name to a usable variant, e.g. `projectM4` or `projectM4-0`.
- Keep the namespace and target names as-is, without any version decorations.
- Create a meta package that has no version number in its package name, e.g. `projectM` as it was before.
- Create additional config scripts in the meta package that use the major and minor version numbers to search the specific version's config dir.
- Perform an additional `find_package(projectM4 ...)` in the meta package to find this package.

Since CMake allows both specifying a minimum version and also EXACT versions, this would still allow using these suffixes in the package name. Qt had this for two versions, but they also replaced it recently with version-less package names, which is _way_ easier to use and maintain.

Reply:
Hi @kblaschke, I have taken the time to look at this in detail now.

> When renaming the package this way, be aware that this will make `find_package` potentially unusable because it doesn't really like dots in the name. this would make CMake-based integration in other projects impossible.

I have changed it to `projectM4` now.

I should note that folder name  `projectM-4.0` does work for `find_package(projectM [..])` technically because of the asterisk in [CMake's search patterns](https://cmake.org/cmake/help/latest/command/find_package.html#search-procedure), e.g. the one in `<prefix>/(lib/<arch>|lib*|share)/cmake/<name>*/` near the end, and e.g. Boost uses this feature consistently:
```
# ls -d1 /usr/lib64/cmake/boost*-[0-9]*.[0-9]/ | wc -l
37
```
No need for that with `projectM4` though, good.

> I'd suggest the following, which other libs also do:
> 
> * Rename the current files/package name to a usable variant, e.g. `projectM4` or `projectM4-0`.

Done. This will save use headaches with projectM 5.

I have pushed a new version now that works well with `find_package` over here. Here's one easy way to see it in action:

```console
$ cd "$(mktemp -d)"
$ git clone --depth 1 --branch fix-collisions-with-projectm-prior-to-4-0 https://github.com/hartwork/projectm
$ cd projectm
$ cmake -S . -B build -DCMAKE_BUILD_TYPE=Release -G Ninja
$ DESTDIR="$PWD"/ROOT/ ninja -C build install

$ cat >hello.c <<HELLO_C_EOF
#include <projectM-4/core.h>
#include <projectM-4/playlist_core.h>

#include <assert.h>
#include <stdio.h>

int main()
{
    projectm_handle pm = projectm_create();
    projectm_playlist_handle playlist = projectm_playlist_create(pm);
    projectm_playlist_destroy(playlist);
    projectm_destroy(pm);
    return 0;
}
HELLO_C_EOF

$ cat >CMakeLists.txt <<CMAKELISTS_TXT_EOF
cmake_minimum_required(VERSION 3.0)

project(hello VERSION 1.0.0)

add_executable(hello hello.c)

find_package(projectM4 REQUIRED CONFIG)
find_package(projectM4Playlist REQUIRED CONFIG)

target_link_libraries(hello PUBLIC libprojectM::projectM libprojectM::playlist)
CMAKELISTS_TXT_EOF

$ cmake -DCMAKE_PREFIX_PATH="${PWD}"/ROOT/usr/local/ -S . -B build_hello
$ make -C build_hello/ VERBOSE=1
$ ./build_hello/hello ; echo $?
```
There is no OpenGL context here, so that program tests compilation and linking more than anything, but that's a detail.

What do you think?